### PR TITLE
fix: selinux tests

### PR DESCRIPTION
**Pull request by fmoral2**
#### Proposed Changes ####

- fix selinux tests that was breaking

#### Testing ####

Checklist:
1. If your PR changes anything on or related to Jenkins, run it pointing to your branch to make sure it's okay.
- no, but will create a new one 

2. Verify code lint; we should not have errors.

 - ok fixed and the error on the lint i am fixing in the  https://github.com/rancher/distros-test-framework/pull/38 
3. Update the documentation if needed.

- n/a 
4. Update makefile and docker run if adding new tests.
- updated 

5. Run your tests at least 4 times with all configurations needed and possible.
- yes both products 

6. If needed test with different os types.
- yes, centos7 , centos8

#### Further Comments ####

tests passing without upgrade and with

### CODE REVIEW

File: pkg/testcase/selinux.go [hunk 1]
```diff
@@ -12,56 +11,36 @@ import (
...
 
 type configuration struct {
 	distroName string
 	cmdCtx
 }
 
-// TestSelinuxEnabled Validates that containerd is running with selinux enabled in the config
-func TestSelinuxEnabled() {
+// setHelper returns the cluster and product
+func setHelper() (*factory.Cluster, string, error) {
```
Comment (mdrahman-suse) [inline pkg/testcase/selinux.go:22]: This func name is confusing. Can we rename to reflect its purpose like `getClusterProduct`

Comment (fmoral2) [reply pkg/testcase/selinux.go:22]: sure

File: pkg/testcase/selinux.go [hunk 2]
```diff
@@ -209,42 +153,35 @@ func selectSelinuxPolicy(product, osType string) cmdCtx {
...
-		log.Println(err)
-	}
-	cluster := factory.AddCluster(GinkgoT())
-	var serverUninstallCmd string
-	var agentUninstallCmd string
-	var serverCmd string
+	cluster, product, err := setHelper()
+	Expect(err).NotTo(HaveOccurred())
+
+	serverUninstallCmd := "sudo rke2-uninstall.sh"
```
Comment (mdrahman-suse) [inline pkg/testcase/selinux.go:177]: can we extract these into shared func? also this will fail if the path to `*-uninstall.sh` is different... as in some OS's it is different: `/usr/local/bin` vs `/opt/local/bin`

Comment (fmoral2) [reply pkg/testcase/selinux.go:177]: why shared if its specific to selinux ?

**Review by mdrahman-suse — COMMENTED** [review]
Few comments

File: pkg/testcase/selinux.go (hunk 2 above)
Comment (mdrahman-suse) [reply pkg/testcase/selinux.go:177]: the uninstall can be re-purposed for other testing, like validating the uninstallation itself, hence I asked if it can be moved...

**Review by mdrahman-suse — DISMISSED** [review]

**Comment by fmoral2** [comment]
` Red Hat Enterprise Linux Server 7.9 (Maipo);`
Screenshot 2023-11-10 at 12 59 37



` CentOS Linux 7 (Core)":`

Screenshot 2023-11-10 at 13 03 40




`  Oracle Linux Server 8.7 : `

Screenshot 2023-11-10 at 13 04 54

File: .golangci.yaml [hunk 3]
```diff
@@ -5,27 +5,26 @@ linters:
...
+  gofmt:
+    simplify: true
   govet:
     check-shadowing: true
   gci:
     local-prefixes: github.com/rancher/distros-test-framework
-  gosec:
-    issues:
-      - G306
-      - G106
```
Comment (VestigeJ) [inline .golangci.yaml:20]: Are these removed because the framework is hitting edge cases and failing to adhere to these rules?

Comment (fmoral2) [reply .golangci.yaml:20]: it was removed not on purpose , going back

**Review by mdrahman-suse — APPROVED** [review]
LGTM

File: shared/aux.go
Comment (VestigeJ) [inline shared/aux.go:1]: Just a few linting fixes it seems there's a rule directly against ending the error return with a newline - not sure why the rule is there but if we're using it we should adhere to it.

**Review by VestigeJ — CHANGES_REQUESTED** [review]
See aux.go for linting fixes - changing the error returns to not send newlines. or we edit the linting rules to ignore this if the output isn't working the way we want.

File: shared/aux.go
Comment (fmoral2) [reply shared/aux.go:1]: this is not an error, just a warn. 

 plus on that place we want to break line to have a better output debugging

**Comment by fmoral2** [comment]
@VestigeJ @mdrahman-suse  , just needed to solve conflicts !

**Review by VestigeJ — DISMISSED** [review]

**Review by VestigeJ — APPROVED** [review]
✅

**Review by mdrahman-suse — APPROVED** [review]
LTGM
